File: app/src/main/java/org/xdty/callerinfo/model/permission/Permission.java
package org.xdty.callerinfo.model.permission;

import android.content.Context;
import android.support.annotation.NonNull;

public interface Permission {

    boolean canDrawOverlays();

    void requestDrawOverlays(Context context, int requestCode);

    int checkPermission(String permission);

    void requestPermissions(Context context, @NonNull String[] permissions, int requestCode);

    boolean canReadPhoneState();

    boolean canReadContact();

}

File: app/src/main/java/org/xdty/callerinfo/model/permission/Permission.kt
package org.xdty.callerinfo.model.permission

import android.content.Context

interface Permission {
    fun canDrawOverlays(): Boolean
    fun requestDrawOverlays(context: Context, requestCode: Int)
    fun checkPermission(permission: String): Int
    fun requestPermissions(context: Context, permissions: Array<String>, requestCode: Int)
    fun canReadPhoneState(): Boolean
    fun canReadContact(): Boolean
}